wazuh/wazuh-dashboard-plugins
# Add vulnerabilities configurations and healthCheck

### Description
Add vulnerabilities configurations and healthCheck
 
### Issues Resolved
- Closes #6146 

### Evidence
<details>

#### Warning message when the vulnerability index does not exist. Allowing to continue.
![01_health_check_without_index_data_WARNING_NOTIFICATION](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/1f1968f7-7820-44a8-ac88-17537522a9f7)

#### Dashboard status when the index does not exist. Each display shows its corresponding message.
![02_vulnerability_detection_without_index_data](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/2350b418-591b-4bbf-9e05-619676af4b4e)

#### Health check when index exists but not mapping
![03_health_check_with_index_data_NO_MAPPING_ON_MANAGER](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/2142c7bc-bbe4-413a-993f-d6122269f573)

![04_health_check_with_index_data_NO_MAPPING_ON_MANAGER_open_logs](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/822673d3-e836-473a-884c-743fa04575b5)

#### Dashboard status when index exists but not mapping
![05_vulnerability_detection_with_index_data_NO_MAPPING_ON_MANAGER](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/bab5c653-e143-4a7d-b303-f37fe276ede6)

#### Health check when index exists and correct mapping
![06_vulnerability_detection_with_index_data_NO_MAPPING_ON_MANAGER_open_logs](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/b2adcc47-73dd-4b29-91bb-583b47dde625)

![07_health_check_with_index_data_MAPPING_ON_MANAGER](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/fc756cc6-9242-4d2c-8b83-5da366dab028)

#### Dashboard status  when index exists and correct mapping
![08_vulnerability_detection_with_index_data_MAPPING_ON_MANAGER](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/472f0331-5d8f-4345-a991-c81f346cf663)

#### Vulnerabilities appSettings
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/5fb5b6b4-a257-4a15-ad04-28f3e8ff9501)

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/acd5fbe9-7f42-434f-9f80-5c75114ff073)



#### Vulnerability index pattern check disabled
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/e52846cf-1bf3-4664-a38f-4530378261b8)


</details>

### Test

_Before remember to create the mapping of port `9200:9200` in `osd1` in the `dev.yml`_
- When raising the environment for the first time. If the vulnerability index does not exist, a Warning should appear in the Health Check. But it should allow you to continue. When going to the vulnerabilities module, the visualizations will show an error because they do not have the index or the mapping.
- Create the index (wazuh-states-vulnerabilities) in the manager and insert data (using script of [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). The health-check will correctly create the index pattern and pass the health-check, although the dashboard will not be seen completely.
- Insert the corresponding mapping and see that the healthcheck passes correctly and the dashboard is shown complete (using dev-tools of [comment](https://github.com/wazuh/wazuh-dashboard-plugins/issues/5763#issuecomment-1779564103)). Refresh vulnerabilities index-pattern then.
- Disable the vulnerability index pattern check in appSettings. In the health-check it should appear as disabled, but it should allow you to continue


### Check List
